### ISSUE TITLE
Feature Request - Support for OPNsense and PFSense firewalls (FreeBSD based)

### ISSUE THREAD

**Issue by LPJon** [amidaware/rmmagent#16]
When do you think it might be possible to include FreeBSD 12 and 13 in the list of supported OS's. Golang is supposed to support FreeBSD but it seems the DEVs of container os-sysinfo is giving a fuss over adding support for it. I was really hoping this might be coming soon as would like to manage several open source firewalls with clients. If it needs payment to move it forward how much are you thinking?

**Comment by redanthrax** [comment]
I have a PR on os-sysinfo but getting traction on it is another thing.

**Comment by LPJon** [comment]
Right!? And they have been bucking it for some time now and not just from you. I did a compiled test with your changes. Its promising but i bet your gonna have to fork them and then make your changes and have tacticalRMM pull the changes from your repo. I think this has been going on since 2019 and they are complaining that there is no reliable way to pull system info for freebsd without using procfs and they refuse to require a user to mount it to get the info. Meshcentral's agent already requires this anyway. Might have to think about switching away from os-info unless you do the changes yourself. Or maybe you could use meshcentrals agent to collect the information for linux based systems. I believe there is already a built in system for that.

**Comment by LPJon** [comment]
Also...on the test i ran...the logging is not working. That's why i opted not to use it. So there might be an issue there too. The freebsd distro is OPNsense that it was tested on.
